### PR TITLE
Format code with CodeFormatter

### PR DESCRIPTION
**Pull request by pmiossec**
Following discussion on #4534

Nuget package: Octokit.CodeFormatter

".\packages\Octokit.CodeFormatter\tools\CodeFormatter.exe" .\GitExtensions.sln /rule-:FieldNames /nounicode /nocopyright

### CODE REVIEW

File: GitUI/Editor/ColorHelper.cs [hunk 1]
```diff
@@ -52,12 +52,12 @@ public static Color MakeColorDarker(Color color)
         public static int GetColorBrightnessIndex(Color c)
         {
             // From: http://www.had2know.com/technology/color-contrast-calculator-web-design.html
-            return (299*c.R + 587*c.G + 114*c.B) / 1000;
+            return (299 * c.R + 587 * c.G + 114 * c.B) / 1000;
```
Comment (drewnoakes) [inline GitUI/Editor/ColorHelper.cs]: Here's one example where I prefer the original formatting as it highlights operator precedence more clearly. The original formatting is something that ReSharper can achieve as part of its cleanup.

File: GitUI/FormProcess.cs [hunk 2]
```diff
@@ -155,12 +154,13 @@ private void processStart(FormStatus form)
...
-                    Process.StandardInput.Write(ProcessInput);
-                    AddMessageLine(string.Format(":: Wrote [{0}] to process!\r\n", ProcessInput));
-*/
+                    throw new NotSupportedException("No non-NULL usages of ProcessInput are currently expected.");  // Not implemented with all terminal variations, so let's postpone until there's at least one non-null case
+
+                    /*
+                                        Thread.Sleep(500);
+                                        Process.StandardInput.Write(ProcessInput);
+                                        AddMessageLine(string.Format(":: Wrote [{0}] to process!\r\n", ProcessInput));
+                    */
```
Comment (drewnoakes) [inline GitUI/FormProcess.cs]: Not a biggie, but this is maybe a bit worse.

File: GitUI/Lemmings.cs [hunk 3]
```diff
@@ -15,15 +15,12 @@ public static Bitmap GetPictureBoxImage(DateTime currentDate)
...
-                    else
-                    {
-                        return Resources.Cow;
-                    }
+                }
+                if (currentDate.Month == 6 && currentDate.Day == 21) //summer
+                {
+                    return Resources.Cow_sunglass;
+                }
+                return Resources.Cow;
```
Comment (drewnoakes) [inline GitUI/Lemmings.cs]: Removed braces here.

**Comment by drewnoakes** [comment]
Quick tip here: If you add `?w=1` to the end of the diff page's URL, then single-line-white-space-only changes are skipped, which reduces the number of lines to review considerably. Still worth skimming a bit without this option to get a sense of how it changes white-space, of course.

**Comment by pmiossec** [comment]
>With respect to this being a solution on an ongoing basis is that the project appears to have been largely abandoned.

Indeed, that's more a one shot solution to fix some inconsistencies using roslyn to guarantee that the 2 source code are equivalent (and follow the coding rules: https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md ).
This is the tool that Microsoft developped to reformat the code of dotnet core before opensourcing it.

They don't use it anymore nor maintain it.
That's just an easy 15min shot ;)

File: GitUI/FormProcess.cs (hunk 2 above)
Comment (pmiossec) [reply GitUI/FormProcess.cs]: I can fix this one easily...

File: GitUI/Lemmings.cs (hunk 3 above)
Comment (pmiossec) [reply GitUI/Lemmings.cs]: I applied manually the Resharper rule to remove the useless `else`

File: GitUI/Editor/ColorHelper.cs (hunk 1 above)
Comment (pmiossec) [reply GitUI/Editor/ColorHelper.cs]: ¯\_(ツ)_/¯

**Comment by pmiossec** [comment]
>Quick tip here:

I know it. I get it from a githubber but I never knew why they don't add it in the web site :(

File: GitUI/Editor/ColorHelper.cs (hunk 1 above)
Comment (drewnoakes) [reply GitUI/Editor/ColorHelper.cs]: Just to clarify, my comment was coming from the perspective of wanting a tool that would run on the code automatically on an ongoing basis.

File: GitUI/FormProcess.cs (hunk 2 above)
Comment (drewnoakes) [reply GitUI/FormProcess.cs]: Again, I was commenting more on the tool than your PR.

**Comment by pmiossec** [comment]
It was more a 5 minutes one shot to show what we could expect. You can close it if you want (or merge it because it fixes a lot of little things).

As you want...

**Comment by RussKie** [comment]
I appreciate you taking time and doing the cleanup.
Right now there are a number of conflicts, I asked whether you wished to continue with the change.

The diff is massive, but looks relatively harmless...

**Comment by drewnoakes** [comment]
This PR has a lot of overlap with #4553 which achieves similar formatting consistency via StyleCop.Analyzers.

**Comment by pmiossec** [comment]
@RussKie I have rebased. 

>The diff is massive, but looks relatively harmless...

Indeed, it was done using Roselyn

@drewnoakes 

>This PR has a lot of overlap with #4553 which achieves similar formatting consistency via StyleCop.Analyzers.

So, see it has a complement because even if #4553 is merged, there is a lot of remaining clean up provided by this PR ;)

**Comment by drewnoakes** [comment]
Indeed. A quick review and it looks good I suspect many of these fixes would be covered by future SCA PRs, so merging this should just make my life easier :)

**Comment by RussKie** [comment]
Thank you

**Pull request merged**